library/constant_time.c

```diff
@@ -79,7 +80,12 @@ static inline uint32_t mbedtls_get_unaligned_volatile_uint32(volatile const unsi
 #if defined(__arm__) || defined(__thumb__) || defined(__thumb2__)
     asm volatile ("ldr %0, [%1]" : "=r" (r) : "r" (p) :);
 #elif defined(__aarch64__)
+#if (SIZE_MAX == 0xffffffff)
```

Comment:
I don't understand how `SIZE_MAX` matters here. Is it a proxy for the size of a pointer? If so, this needs at the very least a comment to explain. And it's wrong for targets with 64-bit pointers but 32-bit `size_t`, which is possible in theory but I don't know if any actually exist on arm. But then why not use `UINTPTR_WIDTH`? It doesn't actually have to match `sizeof(unsigned char *)`, but I think it's less likely to diverge if the choice is 32-bit vs 64-bit.

Reply:
I don't think `UINTPTR_WIDTH` is in C99 (although we only care about __GNUC__ compilers here so maybe that's OK). `UINTPTR_MAX` could work.

Reply:
WIDTH/MAX: yes, I realized immediately after posting.

Also a minor thing: these two lines should be `#elif`.